### PR TITLE
samples: nrf9160: modem_shell: remove usage of CONFIG_NET_PPP_UART_NAME

### PR DESCRIPTION
**Pull request by gmarull**
PPP UART is now chosen in Devicetree, NET_PPP_UART_NAME has been
removed.

### CODE REVIEW

File: samples/nrf9160/modem_shell/src/ppp/ppp_settings.c [hunk 1]
```diff
@@ -33,32 +34,26 @@ static int ppp_settings_uart_dev_conf_get(struct uart_config *uart_conf)
...
-		ret = uart_config_get(ppp_uart_dev, uart_conf);
-		if (ret != 0) {
-			mosh_warn("Cannot get PPP uart config, ret %d", ret);
-		}
-	} else {
-		mosh_warn("No PPP uart device with name %s", CONFIG_NET_PPP_UART_NAME);
-		ret = -1;
+	ret = uart_config_get(ppp_uart_dev, uart_conf);
+	if (ret != 0) {
+		mosh_error("Cannot get PPP uart config, ret %d", ret);
```
Comment (anangl) [inline samples/nrf9160/modem_shell/src/ppp/ppp_settings.c:39]: Why `mosh_warn` is changed to `mosh_error` here? There is another call to `mosh_error` done when the above function returns a non-zero value, so I guess this `mosh_warn` here was intentional.
Same story in `ppp_settings_uart_dev_conf_set`.

Comment (gmarull) [reply samples/nrf9160/modem_shell/src/ppp/ppp_settings.c]: converted back to `mosh_warn`

**Review by anangl — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@gmarull the sdk-zephyr PR is merged; please update this

**Comment by trantanen** [comment]
From modem shell side, @jhirsi has been doing all PPP related developments. He will be back from vacation on 4th August.
Do we have pressure to merge this before that?

**Comment by gmarull** [comment]
> From modem shell side, @jhirsi has been doing all PPP related developments. He will be back from vacation on 4th August. Do we have pressure to merge this before that?

Not really, but note that there are no functional changes.

**Comment by trantanen** [comment]
> > From modem shell side, @jhirsi has been doing all PPP related developments. He will be back from vacation on 4th August. Do we have pressure to merge this before that?
> 
> Not really, but note that there are no functional changes.

Has PPP been tested with modem shell after the changes? If not, I recall we should have a basic test for PPP in our test automation but I'd need our test engineers to run it when they are back on 1st August.

**Comment by rlubos** [comment]
@gmarull @trantanen Guys, this PR blocks other pull requests that involve `sdk-zephyr` updates (here's an example: https://github.com/nrfconnect/sdk-nrf/pull/8263, the CI fails on `mosh` build). Could we get this integrated and do the testing later?

**Comment by gmarull** [comment]
+1 for me, this PR is low risk IMO.

**Comment by trantanen** [comment]
> @gmarull @trantanen Guys, this PR blocks other pull requests that involve `sdk-zephyr` updates (here's an example: #8263, the CI fails on `mosh` build). Could we get this integrated and do the testing later?

I'll let you decide on how you want to proceed. Generally I'm not particularly keen on merging changes to master that haven't been tested even manually (I understood that PPP with these changes is not tested at all).

**Comment by trantanen** [comment]
> > @gmarull @trantanen Guys, this PR blocks other pull requests that involve `sdk-zephyr` updates (here's an example: #8263, the CI fails on `mosh` build). Could we get this integrated and do the testing later?
> 
> I'll let you decide on how you want to proceed. Generally I'm not particularly keen on merging changes to master that haven't been tested even manually (I understood that PPP with these changes is not tested at all).

Hmmm, if I'm not looking at the test results somehow wrongly, CI for this PR is running one PPP test case which passes fine. So I think we are good to go from that perspective.

**Review by rlubos — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Pull request merged**